src/command.rs

```diff
@@ -0,0 +1,16 @@
+use async_trait::async_trait;
+
+#[async_trait(?Send)]
+pub trait Command<Message> {
+    async fn run(&self) -> Message;
```

Comment:
A command is basically just an async function that results in a message that will be fed back into the main loop.